lib/Doctrine/ORM/Cache/Persister/Collection/AbstractCollectionPersister.php

```diff
@@ -171,39 +170,4 @@ public function loadCriteria(PersistentCollection $collection, Criteria $criteri
     {
         return $this->persister->loadCriteria($collection, $criteria);
     }
-
-    /**
-     * Clears cache entries related to the current collection
-     *
-     * @deprecated This method is not used anymore.
-     */
-    protected function evictCollectionCache(PersistentCollection $collection): void
-    {
-        $key = new CollectionCacheKey(
-            $this->sourceEntity->rootEntityName,
-            $this->association['fieldName'],
-            $this->uow->getEntityIdentifier($collection->getOwner())
-        );
-
-        $this->region->evict($key);
-
-        $this->cacheLogger?->collectionCachePut($this->regionName, $key);
-    }
-
-    /**
-     * @deprecated This method is not used anymore.
-     *
-     * @psalm-param class-string $targetEntity
-     */
-    protected function evictElementCache(string $targetEntity, object $element): void
-    {
-        $targetPersister = $this->uow->getEntityPersister($targetEntity);
-        assert($targetPersister instanceof CachedEntityPersister);
-        $targetRegion = $targetPersister->getCacheRegion();
-        $key          = new EntityCacheKey($targetEntity, $this->uow->getEntityIdentifier($element));
-
-        $targetRegion->evict($key);
-
-        $this->cacheLogger?->entityCachePut($targetRegion->getName(), $key);
-    }
```

Comment:
Their removal should be mentioned in UPGRADE.md. It doesn't seem like 2.12.x mention a deprecation either.

Reply:
The methods looked like internal code that is just not used anymore. But you are right, since this class is meant to be extended, I probably should document the deprecation properly. #9511

Reply:
Better safe than BC complaints :-)